sklego/preprocessing/dictmapper.py

```diff
@@ -0,0 +1,48 @@
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from functools import partial
+
+
+class DictMapper(TransformerMixin, BaseEstimator):
+    """
+    Map the values of values of columns according to the input dictionary,
+    fall back to the default if the key is not present in the dictionary.
+
+    :param mapper: The dictionary containing the mapping of the values
+    :param default: The value to fall back to if the value is not in the mapper
+    """
+
+    def __init__(self, mapper, default):
+        self.mapper = mapper
+        self.default = default
+
+    def fit(self, X, y=None):
+        X = check_array(
```

Comment:
Should we maybe check here if X is a single column? If multiple columns are passed I think this object should break.

Reply:
I agree a single column would be the typical use case. I can also imagine use cases where we would like to use it one multiple columns:

```
  value_before value_after
0          foo         bar
```

Reply:
Fair enough.